File: elabadu/src/services/UserService.js
import axios from 'axios'

const USER_BASE_REST_API_URL = 'http://localhost:8080/api/v1';
class UserService {

    saveUser(user) {
        return axios.post(USER_BASE_REST_API_URL + "/save", user);
    }

    getAllUser() {
        return axios.get(USER_BASE_REST_API_URL + "/");
    }

    getUserById(id) {
        return axios.get(USER_BASE_REST_API_URL + "/" + id);
    }

    deleteUser(id) {
        return axios.delete(USER_BASE_REST_API_URL + "/delete/" + id);
    }

    updateUser(id, user) {
        return axios.put(USER_BASE_REST_API_URL + "/update/" + id, user);
    }
}
export default new UserService();

File: user backend/user backend/usermetadata/src/main/java/com/example/usermetadata/service/UserService.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.example.usermetadata.service;

import com.example.usermetadata.data.UserMetaData;
import com.example.usermetadata.data.UserRepo;
import java.util.List;
import java.util.Optional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

/**
 *
 * @author User
 */
@Service
public class UserService implements UserServiceCo{
    @Autowired
	private UserRepo userRepo;

    @Override
    public UserMetaData createUser(UserMetaData user) {
        return userRepo.save(user);
    }

    @Override
    public List<UserMetaData> getAllUser() {
        return userRepo.findAll();
    }

    @Override
    public UserMetaData getUserById(int id) {
       return (UserMetaData) userRepo.findById(id);
    }

    @Override
    public void deleteUser(int id) {
         UserMetaData user = (UserMetaData) userRepo.findById(id);
		if (user != null) {
			userRepo.delete(user);
		}
    }

    
@Override
    public UserMetaData updateUser(int id, UserMetaData user) {
         return userRepo.save(user);
    }
    
}
